core/iwasm/common/wasm_c_api.c

```diff
@@ -4886,22 +4886,22 @@ wasm_instance_new_with_args(wasm_store_t *store, const wasm_module_t *module,
         func_host = wasm_extern_as_func(in);
         /* it is a placeholder and let's skip it*/
         if (!func_host->type) {
-            func_import++;
             continue;
         }
+        if (func_host->func_idx_rt < 0)
+            continue;
 
-        func_import->with_env_arg = func_host->with_env;
+        CApiFuncImport *linkFunc = func_import + func_host->func_idx_rt;
```

Comment:
the problem is `funx_idx_rt` is not predictable when people work with host functions. This is also why `wasm_func_new()` assigns `-1` to this field.